autowarefoundation/autoware.universe
# feat(mission_planner)!: introduce route_selector node

## Description

Introduce the route_selector node to the mission_planner package. The route_selector node manages main/mrm routes and makes requests to the mission_planner node. Therefore, now the mission_planner node only needs to plan the current route.

## Related links

- Related to https://github.com/autowarefoundation/autoware.universe/issues/6319
- Related to https://github.com/autowarefoundation/autoware.universe/pull/6362
- Related to https://github.com/autowarefoundation/autoware.universe/pull/6364
- Depends on https://github.com/tier4/tier4_autoware_msgs/pull/111

## Tests performed

1. Merge the dependent PRs.
2. Launch planning simulator.
3. Add MrmGoalTool and RouteSelectorPanel to rviz.
4. Set topic name from rviz tool property panel.
    - 2D Goal Pose: `/rviz/route_selector/main/goal`
    - Mrm Goal Pose: `/rviz/route_selector/mrm/goal`
5. Check if the normal route can be set (use 2D Goal Pose)
6. Check if the mrm route can be set (use Mrm Goal Pose)
7. Check if it can return to normal route (push clear button of MRM state)
8. Check if the mrm route can be set again (use Mrm Goal Pose)
9. Check if the normal route setting is rejected during MRM (use 2D Goal Pose)

https://evaluation.tier4.jp/evaluation/reports/7d8458ad-5132-5f28-b0b2-1c117bc1b7e6?project_id=prd_jt

## Notes for reviewers

https://github.com/autowarefoundation/autoware.universe/assets/43976882/b841c383-f30b-41fd-9d48-62e7facdeced

## Interface changes

Add:
- /planning/mission_planning/route_selector/main/route
- /planning/mission_planning/route_selector/main/state
- /planning/mission_planning/route_selector/mrm/route
- /planning/mission_planning/route_selector/mrm/state
- /planning/mission_planning/mission_planner/clear_route
- /planning/mission_planning/mission_planner/set_lanelet_route
- /planning/mission_planning/mission_planner/set_waypoint_route
- /planning/mission_planning/route_selector/main/clear_route
- /planning/mission_planning/route_selector/main/set_lanelet_route
- /planning/mission_planning/route_selector/main/set_waypoint_route
- /planning/mission_planning/route_selector/mrm/clear_route
- /planning/mission_planning/route_selector/mrm/set_lanelet_route
- /planning/mission_planning/route_selector/mrm/set_waypoint_route

Type change:
- /planning/mission_planning/state

## Effects on system behavior

The MRM route will no longer be reflected in AD API.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
